kernel/Utilities/GlobalMatrix.h

```diff
@@ -118,22 +121,47 @@ class GlobalPetscMatrix : public GlobalMatrix {
 
     GlobalPetscMatrix(const GlobalIndexing& rowIndexing,
                       const GlobalIndexing& columnIndexing, int elementMatrixID,
+                      int faceMatrixID = -1)
+        : GlobalPetscMatrix(
+              rowIndexing, columnIndexing,
+              Table2D<bool>(rowIndexing.getNumberOfIncludedUnknowns(),
+                            columnIndexing.getNumberOfIncludedUnknowns(),
+                            faceMatrixID >= 0),
+              elementMatrixID, faceMatrixID){};
+    GlobalPetscMatrix(const GlobalIndexing& rowIndexing,
+                      const GlobalIndexing& columnIndexing,
+                      Table2D<bool> faceCoupling, int elementMatrixID,
                       int faceMatrixID = -1);
 
     ~GlobalPetscMatrix() override;
 
     void assemble() override;
 
-    void reinit() override;
+    void reinit() override { reinit(true); }
+    /// Reinitialize the matrix
+    /// \param reinitFaceCoupling Whether to reinit the face coupling
+    void reinit(bool reinitFaceCoupling);
```

Comment:
I don't think there is any use, you either reinit it to the default value, or you keep the old one. I have no foresight about any future values and the direction, so that would make naming it be quite hard.

So any suggestions would be welcome.

Reply:
yeah I have no good idea either. Then leave it as it is.